src/modules/flow/http-client/http-client.c

```diff
@@ -0,0 +1,600 @@
+        SOL_WRN("Failed to keep pending connection.");
+        sol_http_client_connection_cancel(connection);
+        return -ENOMEM;
+    }
+
+    return 0;
+}
+
+/*
+ * --------------------------------- irange node -----------------------------
+ */
+static void
+int_request_finished(void *data,
+    const struct sol_http_client_connection *connection,
+    struct sol_http_response *response)
+{
+    struct sol_flow_node *node = data;
+    struct http_data *mdata = sol_flow_node_get_private_data(node);
+    struct sol_irange irange;
+
+    if (check_response(mdata, node, connection, response) < 0)
+        return;
+
+    if (streq(response->content_type, "application/json")) {
+        struct sol_json_scanner scanner;
+        struct sol_json_token token, key, value;
+        enum sol_json_loop_reason reason;
+
+        sol_json_scanner_init(&scanner, response->content.data, response->content.used);
+        SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+            struct sol_json_scanner sub_scanner;
+            if (check_key(mdata->url, &key))
+                continue;
+            sol_json_scanner_init(&sub_scanner, value.start, value.end - value.start);
+            SOL_JSON_SCANNER_OBJECT_LOOP (&sub_scanner, &token, &key, &value, reason) {
+                if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "value")) {
+                    if (sol_json_token_get_int32(&value, &irange.val) < 0)
+                        goto error;
+                } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "min")) {
+                    if (sol_json_token_get_int32(&value, &irange.min) < 0)
+                        goto error;
+                } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "max")) {
+                    if (sol_json_token_get_int32(&value, &irange.max) < 0)
+                        goto error;
+                } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "step")) {
+                    if (sol_json_token_get_int32(&value, &irange.step) < 0)
+                        goto error;
+                }
+            }
+            sol_flow_send_irange_packet(node,
+                SOL_FLOW_NODE_TYPE_HTTP_CLIENT_INT__OUT__OUT, &irange);
+            return;
+        }
+    } else {
+        errno = 0;
+        irange.val = strtol(response->content.data, NULL, 0);
+        if (errno)
+            goto error;
+        sol_flow_send_irange_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_CLIENT_INT__OUT__OUT, &irange);
+        return;
+    }
+
+error:
+    sol_flow_send_error_packet(node, -EINVAL,
+        "%s Could not parser the url's contents ", mdata->url);
+}
+
+static int
+int_post_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,
+    const struct sol_flow_packet *packet)
+{
+    int r;
+    struct sol_irange value;
+    struct sol_http_param params;
+    struct http_data *mdata = data;
+    struct sol_http_client_connection *connection;
+    char min[100], max[100], val[100], step[100];
```

Comment:
400 bytes is a little bit too much for integer representation... `3 * sizeof(value.val)` should be a good approximation that doesn't waste too much too space.